libraries/lib-preferences/Prefs.cpp

```diff
@@ -233,49 +233,76 @@ void FinishPreferences()
    }
 }
 
-SettingScope *SettingScope::sCurrent = nullptr;
+namespace
+{
+thread_local std::vector<SettingScope*> sScopes;
```

Comment:
Why bother with `thread_local`?

We only need this now for the main thread, and if we ever really wanted to make it work correctly with multiple threads, there would be more work to make it behave right, like checking for the disjointness of the sets of settings objects updated by concurrent scopes, making commit fail otherwise for the loser of the race, writing uses of the transaction scope to detect failure and retry, etc.

Reply:
There isn't even any protection against races to update a single Settings object even without this "transaction" system.

So really then, there should be a comment in the header that transations (and in fact all writes to settings) are intended for main thread only.  UB if you try otherwise.

Reply:
Let's drop it, no problems